hw/xbox/nv2a/nv2a_pgraph.c

```diff
@@ -4089,31 +4089,113 @@ static uint8_t* convert_texture_data(const TextureShape s,
                                      unsigned int slice_pitch)
 {
     //FIXME: Handle all formats
-    if (s.color_format == NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8R8G8B8) {
+    if ((s.color_format == NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8R8G8B8) ||
```

Comment:
FIXME: Explain what this code path does

Note to self: This flips the sign.

Normally this is two's-complement. So:

- `0x80 = -128 = 128`
- `0xFF = ..-1 = 255`
- `0x00 =  ...0 = ..0`
- `0x7F =  .127 = 127`

Note the discontinuity between 0xFF and 0x00 when interpreting it as unsigned value. By flipping the sign bit and interpreting as unsigned, we achieve the following:

- `0x80` will be `0x00 =  ...0 = ..0`
- `0xFF` will be `0x7F =  .127 = 127`
- `0x00` will be `0x80 = -128 = 128`
- `0x7F` will be `0xFF = ..-1 = 255`

\- So we have gotten rid of the discontinuity.
In the shader, we scale and add a bias, to bring it back into the -128 to 127 range.